packages/patterns/src/getOpaquePattern.ts

```diff
@@ -29,8 +34,11 @@ function getOpaquePattern(pattern) {
 
 export default getOpaquePattern;
 
-function objectWithout(obj, propertiesToRemove) {
-  return Object.keys(obj)
-    .filter((key) => !propertiesToRemove.includes(key))
-    .reduce((newObj, key) => ({ ...newObj, [key]: obj[key] }), {});
+function objectWithout(
+  obj: Record<string, unknown>,
+  propertiesToRemove: Array<string>
+) {
+  return Object.fromEntries(
+    Object.entries(obj).filter(([key]) => !propertiesToRemove.includes(key))
+  );
```

Comment:
I made a slight improvement here - this is way faster than re-creating objects over and over.